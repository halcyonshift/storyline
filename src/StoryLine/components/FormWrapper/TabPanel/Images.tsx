import ImageList from '@mui/material/ImageList'
import ImageListItem from '@mui/material/ImageListItem'
import Image from '@sl/components/Image'
import { TabPanelProps } from './types'

const ImagesPanel = ({ notes }: TabPanelProps) => {
    return (
        <ImageList cols={3} className='m-1'>
            {notes.map((note) => (
                <ImageListItem key={`image-${note.id}`}>
                    <Image path={note.image} alt={note.title} loading='lazy' />
                </ImageListItem>
            ))}
        </ImageList>
    )
}

export default ImagesPanel
