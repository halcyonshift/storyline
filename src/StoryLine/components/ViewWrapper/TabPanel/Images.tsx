import { Box } from '@mui/material'
import Gallery from '@sl/components/Gallery'
import { ImagesPanelProps } from './types'

const ImagesPanel = ({ images }: ImagesPanelProps) => (
    <Box className='p-1'>
        <Gallery images={images} />
    </Box>
)

export default ImagesPanel
