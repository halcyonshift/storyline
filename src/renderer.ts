import { init } from '@sentry/electron/renderer'
import { init as reactInit } from '@sentry/react'

if (parseInt(process.env.DEBUG) === 0) {
    init({ dsn: process.env.SENTRY }, reactInit)
}

import '@fontsource/roboto/300.css'
import '@fontsource/roboto/400.css'
import '@fontsource/roboto/500.css'
import '@fontsource/roboto/700.css'
import './index.css'
import './StoryLine'
