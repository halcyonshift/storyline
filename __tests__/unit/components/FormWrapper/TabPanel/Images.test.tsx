import renderer from 'react-test-renderer'
import Images from '@sl/components/FormWrapper/TabPanel/Images'

describe('<TabPanel.Images />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Images notes={[]} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
