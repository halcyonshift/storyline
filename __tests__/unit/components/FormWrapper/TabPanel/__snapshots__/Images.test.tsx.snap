// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPanel.Images /> renders correctly 1`] = `
<ul
  className="MuiImageList-root MuiImageList-standard m-1 css-rvv586-MuiImageList-root"
  style={
    {
      "gap": 4,
      "gridTemplateColumns": "repeat(3, 1fr)",
    }
  }
/>
`;
